spec.html

```diff
@@ -28095,40 +28102,41 @@ <h1>Week Day</h1>
-        <p>If an implementation does not support a conversion described above or if political rules for time _t_ are not available within the implementation, the result must be 0.</p>
+        <p>LocalTZA is an implementation-defined abstract operation that returns the positive or negative local time zone UTC offset (in milliseconds) at a particular instant indicated by its arguments. The result should be dependent upon political rules regarding standard time and daylight saving time in effect at that instant.</p>
         <emu-note>
           <p>It is recommended that implementations use the time zone information of the IANA Time Zone Database <a href="https://www.iana.org/time-zones/">https://www.iana.org/time-zones/</a>.</p>
-          <p>1:30 AM on November 5, 2017 in America/New_York is repeated twice (fall backward), but it must be interpreted as 1:30 AM UTC-04 instead of 1:30 AM UTC-05. LocalTZA(TimeClip(MakeDate(MakeDay(2017, 10, 5), MakeTime(1, 30, 0, 0))), false) is <emu-eqn>-4 &times; msPerHour</emu-eqn>.</p>
-          <p>2:30 AM on March 12, 2017 in America/New_York does not exist, but it must be interpreted as 2:30 AM UTC-05 (equivalent to 3:30 AM UTC-04). LocalTZA(TimeClip(MakeDate(MakeDay(2017, 2, 12), MakeTime(2, 30, 0, 0))), false) is <emu-eqn>-5 &times; msPerHour</emu-eqn>.</p>
-          <p>Local time zone offset values may be positive <i>or</i> negative.</p>
+        </emu-note>
+        <p>When _isUTC_ is *true*, <dfn>LocalTZA( _t_, *true* )</dfn> interprets _t_ as a time value representing an instant in time. It returns the offset of the local time zone from UTC in milliseconds at that instant. When the result is added to _t_, it should yield the time value for which UTC date and time of day are equal to the local date and time of day at time value _t_.</p>
+        <p>When _isUTC_ is *false*, <dfn>LocalTZA( _t_, *false* )</dfn> interprets _t_ as a time value representing a date and time of day. It returns the offset of the local time zone from UTC in milliseconds at the smallest (closest to negative infinity) time value _earliestMatch_ for which either the local date and time of day at _earliestMatch_ is equal to the UTC date and time of day at _t_ or the local date and time of day at _earliestMatch_ + 1 is greater than the UTC date and time of day at _t_. When the result is subtracted from _t_, it should yield the smallest time value for which the local date and time of day are equal to the UTC date and time of day at _t_. If no such time value exists because of local times skipped by positive time zone transitions (e.g. when the daylight saving time starts or the time zone adjustment is increased due to a time zone rule change), then subtracting the result from _t_ should instead yield the time value for which local date and time of day in the UTC offset applicable before the transition are equal to the UTC date and time of day at _t_.</p>
```

Comment:
Can this be given in pseudo code?
1. Get the list of all time zone transitions for a full supported date range (the timestamps, for which LocalTZA(t, true) != LocalTZA(t, true).
2. iterate the list in sorted order, 
2.1. if localtime < transitionTimestamp + max(LocalTZA(transitionTimestamp - 1), LocalTZA(transitionTimestamp))
2.1.1. return localtime - LocalTZA(transitionTimestamp - 1);
3. return localtime - LocalTZA(<the timestamp of the last transition from the list>, true)

Reply:
@gibson042 
It is the same expression as you are using:
FInd smallest integer `_earliestMatch_` for which `LocalTime(_earliestMatch_) === _t_ || LocalTime(_earliestMatch_ + 1) > _t_`
Return localtime - LocalTime(_earliestMatch_)

where `LocalTime` is the abstract operation mentioned in the spec.

Reply:
Hmm, I'd rather not introduce cyclic references between the text of isUTC and LocalTime.

Reply:
@gibson042 , it would be better to remove the LocalTZA so and have LocalTime be the thing doing what LocalTZA(t, false) is doing now, but that will be a bigger change.

Reply:
@gibson042 , it is not cyclic, the LocalTime uses LocalTZA(t, true) branch